### DISCUSSION TITLE
Basic CC Switch Entity Still Exists After Re-Interview: HomeSeer HS-FS100+

### DISCUSSION THREAD

**Discussion by The00Dustin** [zwave-js/zwave-js#6594] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Basic CC Switch entities are showing up in HA for the HomeSeer Flex Sensor HS-FS100+ and persist after re-interview.

What did you expect to happen instead?
I expected these entities to disappear, as I believe they are related to https://community.home-assistant.io/t/psa-upcoming-z-wave-js-changes-and-z-wave-certification-how-it-impacts-you/638288#what-probably-brought-you-here-basic-cc-light-3

### Software versions

Driver (node-zwave-js): 12.4.1

Z-Wave JS UI: 9.6.2

Home Assistant Z-Wave Integration: 1.33.0

Home Assistant Z-Wave JS Addon: 3.1.0

### Device information

Manufacturer: HomeSeer
Model name: HS-FS100+
Node ID: 65 & 69

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

This log contains 3 re-interviews between the two devices at 10:51 & 10:53 UTC:
[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/13861204/zwave-js-ui-store.zip)
The device that was re-interviewed twice was re-interviewed twice because the configuration v3 section disappeared after the first re-interview.  These two devices have neither ever populated the values of the configuration parameters after re-interview, so I manually re-enter the values each time.  I did not re-enter the value for parameter 3 prior to capturing logs because I forgot about this and didn't make note of its (non-default) value.  I did set them again after collecting the logs since reviewing the logs made it easy to see what setting was in use (240).

**Comment by The00Dustin** [comment]
These devices are somewhat confusing in that once the sensor has a detection the value that HA exposes by default (a binary sensor) is flipped on permanently until it is reset by idling Notification v8 [65-113-0-Light sensor-Light detection status] (there is a diagnostic entity exposed to do this which is visible by default), but there is no binary sensor visible in Z-Wave JS UI that stays on permanently, only a binary switch [65-37-0-currentValue], so presumably the binary sensor in HA reflects the state of that binary switch in Z-Wave.  Meanwhile, this switch entity in HA does not reflect the state of the binary switch in Z-Wave, and presumably should not be created, however, I just noticed a second switch in Z-Wave JS UI, [65-37-0-targetValue], but I'm not sure what it's for, best guess is activate/deactivate the buzzer (which is disabled by default, possibly meaning the switch will do nothing by default), I can test this later on and report back.

While on the topic of this device's config, it may also be worth pointing out that the binary sensor that is visible in Z-Wave JS UI [65-48-0-Any], which reflects the current state of the sensor at all times, is hidden by default in HA (I've actually enabled it and disabled the other for my use cases).  I'm  not sure why this sensor is hidden by default or if it should be, nor am I sure if the node config has anything to do with that.

While trying to find the answer to the question about that second binary switch, I also noticed that the manual link in the db is 404.  I found this link with a search: https://homeseer.com/wp-content/uploads/2020/09/HS-FS100-Manual-4.pdf  I couldn't find that link on the HomeSeer site, though, and I found this one there: https://docs.homeseer.com/products/hs-fs100-user-guide  Since the config may need other edits and I'm not sure which manual should be linked to, I'm providing the links here instead of opening a separate PR.

**Comment by The00Dustin** [comment]
OK, so I tested the basic switch entity via Z-Wave JS UI, and it sounds the buzzer when I turn it on manually, it also shows that I did that in HA's logbook.  Presumably this switch would turn on automatically along with the other switch if configuration parameter 4 was set to 1.  Given two binary switches (one read-only) and one binary sensor in Z-Wave (3 values total) which seem to correctly match up with two binary sensors and one switch in HA (three entities total), I suspect I originally assumed that the switch was an issue based on timing (having added these devices just after learning that all the other unexpected basic set entities showed up with sensors) and didn't even try to test it.  That having been said, it is likely that one or two changes are needed for this devices config file even though they don't match with the title of this discussion.  I'll list them below.  Please let me know if you agree that the first item is an issue, but it doesn't lie with the config file, and in that case, where I should open a new issue for it:

- The actual binary sensor is hidden by default in HA
- The link to the manual leads to a 404

**Reply by kpine** [reply]
> The actual binary sensor is hidden by default in HA

That's not a bug, it's an intentional design decision. HA hides Binary Sensor CC based entities by default for Notification Sensor device types because it's a "legacy CC" (not recommended for implementation). Instead, HA expects the device will primary implement Notification CCs instead as it's a requirement for that device type.

In practice, this expectation is not very consistently realized. Some devices provide Binary Sensor but it is disabled on config by default; AFAIK this is the motivation for the design as the sensors are not useful and it's confusing to users. Other Notification Sensor devices do the opposite (even though it's wrong by my understanding): support Binary Sensor by default, and the Notification CC sensors are enabled by config, so the current behavior is confusing in this case. There's a third class of devices like this HomeSeer, where both are provided by default and one is more convenient than the other, and finally a fourth class of device where binary sensor is used because Notification doesn't support the notification type (e.g. Zooz shock sensors).

Based on my experience, I think you will have a tough time justifying a change of this design to HA core developers though. This problem is only solvable in a way that makes everyone happy by customizing the behavior per-device, which is a burden that HA doesn't want to take on when possible.

> These devices are somewhat confusing in that once the sensor has a detection the value that HA exposes by default (a binary sensor) is flipped on permanently until it is reset by idling Notification v8 [65-113-0-Light sensor-Light detection status]

If this device is supporting Notification CC **v8**, and the state never goes back to idle (0) without manual intervention (is that what you mean?), then it sounds like a bug in the device. v8  added a requirement that notification state variables with an idle state must be idled, and the Light detection status variable one of these kind of state variables

![image](https://github.com/zwave-js/node-zwave-js/assets/791794/9f795d8e-5bfe-4df5-a0e2-b9f007783116)

**Reply by The00Dustin** [reply]
> If this device is supporting Notification CC v8, and the state never goes back to idle (0) without manual intervention (is that what you mean?), then it sounds like a bug in the device. v8 added a requirement that notification state variables with an idle state must be idled, and the Light detection status variable one of these kind of state variables

It shows Notification v8 in Z-Wave JS UI with `[65-113-0-Light sensor-Light detection status]` currently showing `[1] Light detected` and an idle button below that.  This is in spite of the fact that the light shut off 37 hours ago and Binary Sensor v1 `[65-48-0-Any] Sensor state (Any)` switched to `false` immediately when that happened.  I guess what I probably did is disabled the notification entity (binary sensor), so I can't see how long it has actually been [1], but I should imagine it has been for much longer than that.  I agree that it does sound like this is a bug in the device.  I can reach out to HomeSeer support on that front.

I am now curious as to whether that is the only bug in the device, though, because I wouldn't expect this behavior when there is no packet loss (but IIRC, node 65 did not have any timeout responses during the re-interview):

> These two devices have neither ever populated the values of the configuration parameters after re-interview, so I manually re-enter the values each time.

Beyond that, since the issue I thought I was reporting doesn't appear to exist and everything else that isn't a firmware bug seems to be as expected except for the bad manual link, I'm wondering if I should close this discussion and submit a PR for the manual link.  However, I'm also not sure if the PDF I found or the web page linked from HomeSeer's current support would be a better link use submit in such a PR...
